docs/tableconfig_schema.rst

```diff
@@ -207,3 +243,116 @@ The ``metadata`` section is used for passing special key-value pairs into Pinot
+
+Here is a minimal example of what the ``streamConfigs`` section may look like:
+
+.. code-block:: none
+
+    "streamConfigs" : {
+      "realtime.segment.flush.threshold.size": "0",
+      "realtime.segment.flush.threshold.time": "24h",
+      "realtime.segment.flush.desired.size": "150M",
+      "streamType": "kafka",
+      "stream.kafka.consumer.type": "LowLevel",
+      "stream.kafka.topic.name": "ClickStream",
+      "stream.kafka.consumer.prop.auto.offset.reset" : "largest"
+    }
+
+
+The ``streamType`` field is mandatory. In this case, it is set to ``kafka``.
+StreamType of ``kafka`` is supported natively in Pinot.
+You can use default decoder classes and consumer factory classes.
+Pinot allows you to use other stream types with their own consumer factory
+and decoder classes (or, even other decoder and consumer factory for ``kafka`` if
+your installation formats kafka messages differently). See :ref:`pluggable-streams`.
+
+There are some configurations that are generic to all stream types, and others that
+are specific to stream types.
+
+Configuration generic to all stream types
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* ``realtime.segment.flush.threshold.size``: Maximum number of rows to consume before persisting the consuming segment.
+
+   Note that in the example above, it is set to
+   ``0``. In this case, Pinot automatically computes the row limit using the value of ``realtime.segment.flush.desired.size``
+   described below. If the consumer type is ``HighLevel``, then this value will be the maximum per consuming
+   segment. If the consumer type is ``LowLevel`` then this value will be divided across all consumers being hosted
+   on any one pinot-server.
+   
+   Default is ``5000000``.
+
+* ``realtime.segment.flush.threshold.time``: Maximum elapsed time after which a consuming segment should be persisted.
+
+   The value can be set as a human readable string, such as ``"1d"``,
+   ``"4h30m"``, etc. This value should be set such that it is not below the retention of
+   messages in the underlying stream, but is not so long that it may cause the server to run out of memory.
+
+   Default is ``"6h"``
+
+
+* ``realtime.segment.flush.desired.size``:  Desired size of the completed segments.
+
+   This setting is supported only if consumer type is set to ``LowLevel``.
+   This value can be set as a human readable string such as ``"150M"``, or ``"1.1G"``, etc.
+   This value is used when ``realtime.segment.flush.threshold.size``
+   is set to 0. Pinot learns and then estimates the number of rows that need to be consumed so that the 
+   persisted segment
+   is approximately of this size. The learning phase starts by setting the number of rows to 100,000
+   (can be changed with the setting ``realtime.segment.flush.autotune.initialRows``).
+   and increasing to reach the desired segment size. Segment size may go over the desired size significantly
+   during the learning phase.
+   Pinot starts corrects the estimation as it
```

Comment:
s/Pinot starts corrects/Pinot corrects ?

Reply:
Changed it to
"Pinot corrects the estimation as it
   goes along, so it is not guaranteed that the resulting completed segments are of the exact size
   as configured. You should set this value to optimize the performance of queries. You should set this value to optimize the performance of queries (i.e. neither
   too small nor too large)"
"